atoma-service/src/streamer.rs

```diff
@@ -563,6 +572,10 @@ impl Streamer {
 
         let (signature, response_hash) = self.sign_chunk(&chunk)?;
 
+        let intra_token_generation_timer = CHAT_COMPLETIONS_INTRA_TOKEN_GENERATION_TIME
```

Comment:
I realized the `intra_token_generation_timer` should not be instantiated in here. Instead, it should be at the end of the current logical flow (current stream poll), when the stream polls next then we should observe time and instantiate a new intra token timer.

Reply:
That makes sense, I've updated it in 7b55718